pilot/pkg/model/context.go

```diff
@@ -86,6 +87,10 @@ type Proxy struct {
 	// NOTE: DO NOT USE THIS FIELD TO CONSTRUCT DNS NAMES
 	ConfigNamespace string
 
+	// NamespaceDependencies contains a list of namespaces that should be used to configure
```

Comment:
add another enum here:
 dependencyMode: which is either AllNamespaces or SameNamespace.

Reply:
What's for? Now if length of `NamespaceDependencies` equals 0, it is  AllNamespaces, otherwise it is the SameNamespace and with other imported namesapces.